test/test.cpp

```diff
@@ -127,4 +127,35 @@ BOOST_AUTO_TEST_CASE( open_and_create ) {
    BOOST_REQUIRE_EQUAL( new_book.b, copy_new_book.b );
 }
 
+BOOST_AUTO_TEST_CASE( oom_flush_dirty_pages ) {
+   temp_directory temp_dir;
+   const auto& temp = temp_dir.path();
+   std::cerr << temp << " \n";
+
+   constexpr size_t db_size = 4ull << 30; // 4GB
+   constexpr size_t max_elems = db_size / (sizeof(book) + 16 + 4);
+   chainbase::database db(temp, database::read_write, db_size, false, pinnable_mapped_file::map_mode::mapped_private);
+   db.add_index< book_index >();
+
+   auto& pmf = db.get_pinnable_mapped_file();
+   pmf.set_oom_threshold(100); // set a low threshold so that we hit it with a 4GB file
+   pmf.set_oom_delay(1);
+
+   size_t flush_count = 0;
+   for (int i=0; i<max_elems; ++i) {
```

Comment:
fwiw there is a miss-matched signed comparison warning here

Reply:
Thanks, Fixed.